qiskit_experiments/warnings.py

```diff
@@ -0,0 +1,274 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A collection of typical warnings."""
+
+import functools
+import warnings
+
+from typing import Callable, Optional, Type, Dict, Union
+
+from qiskit_experiments.framework import BaseExperiment, BaseAnalysis
+
+
+def deprecated_function(
+    last_version: Optional[str] = None,
+    msg: Optional[str] = None,
+    stacklevel: int = 2,
+) -> Callable:
+    """A function or method decorator to show deprecation warning.
+
+    Args:
+        last_version: The Qiskit Experiment version that this function is removed.
+        msg: Extra message, for example, to indicate alternative approach.
+        stacklevel: Stacklevel of this warning. See Python Warnings documentation for details.
+
+    Examples:
+
+        .. code-block::
+
+            @deprecated_logic(last_version="0.3", msg="Use new_function instead.")
```

Comment:
```suggestion
            @deprecated_function(last_version="0.3", msg="Use new_function instead.")
```
code <-> documentation mismatch.